arangod/Auth/TokenCache.cpp

```diff
@@ -157,12 +157,12 @@ auth::TokenCache::Entry auth::TokenCache::checkAuthenticationBasic(std::string c
   {
     WRITE_LOCKER(guard, _basicLock);
     if (authorized) {
-      if (!_basicCache.emplace(secret, entry).second) {
+      if (!_basicCache.try_emplace(secret, std::move(entry)).second) {
```

Comment:
This needs to be reverted, because it makes tests fail.

Reply:
Can't this whole `if` be simplified? With C++17 we can use `insert_or_assign` - no need to try and erase in case the key already exists. This also allows us to `move` the secret:
```
_basicCache.insert_or_assign(std::move(secret), std::move(entry));
```

Reply:
Exactly. `secret` can be moved, `entry` must not be moved.
insert_or_assign is exactly the right thing to do.

Reply:
Just wondering why pre 17 `operator[]` was not used.

Reply:
github does not update

Reply:
Because sometimes the simply solution is just _too_ obvious... 😉